cost.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package celvet
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/google/cel-go/cel"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	schemacel "k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel"
+)
+
+type costError struct {
+	Program cel.Program
```

Comment:
My original thinking here was that exporting `Program` would allow another separate function to take the AST and make the cost breakdown visualization - I probably should have made that more clear in the design, sorry.

Reply:
If the function you plan will be in this same package, the field does not need to be exported. If the function is in a different package how will it access a field of an unexported type?

Reply:
Oh, good point. I've removed it for the time being (no code actively uses that field at present) but when I add it back it will be unexported.

Reply:
Leaving this unresolved but I did want to point out `gopls` gave a linter warning on having `costError` unexported despite being used as a return value for an exported function, so I've exported `CostError`.